### PR TITLE
OSD custom string

### PR DESCRIPTION
**Pull request by aubreyrjones**
This patch adds a user-defined string which can be displayed in the OSD but which has no other semantics defined within INAV.

With user-defined semantics, I expect many emergent use cases. But the motivating use case for this string is displaying an amateur radio callsign to satisfy US station identification requirements. As a callsign is assigned to a human operator and shared among all FPV transmitters they control, co-opting the craft name is not a viable solution to this problem. The craft name is used by some software and many humans to uniquely identify that craft within a user's collection of craft, and concatenating craft name with callsign wastes a lot of space in the OSD with the legally-irrelevant craft name.

This patch set includes implementation of the OSD element itself, 16 characters of storage for the string, the cli command `osd_custom_string` which sets or retrieves the string value, and a pair of MSP commands for setting and retrieving the string through the configurator.

### CODE REVIEW

File: src/main/io/osd.c
Comment (shellixyz) [inline src/main/io/osd.c]: Since the string length is not used this could be optimized by replacing with `if (osdConfig()->customString[0] == '\0')`

File: src/main/fc/fc_msp.c
Comment (shellixyz) [inline src/main/fc/fc_msp.c]: Please add empty lines before and after the case blocks

File: src/main/io/osd.c
Comment (shellixyz) [inline src/main/io/osd.c]: I think this could be optimized by replacing with `osdConfig->customString[0] = '\0'`. It could optimize the code space usage.

**Review by shellixyz — CHANGES_REQUESTED** [review]
Looking good to me. Please rebase.

**Comment by fiam** [comment]
This would be so much easier to understand for users if it was called `pilot_name`. Functionally, it seems all right.

**Comment by aubreyrjones** [comment]
@fiam The goal is to have a user-settable field which does not have *any* meaning defined by INAV developers. This serves the same purpose as a `void *userData` member of an API-defined struct. It is a place for users, not the API developers, to define utility. If you called it `userCallbackPtr`, users would feel free only to put a callback pointer in it, since the API developers' intent is clear.

Referring to the field as "pilot name" defines semantics. Some future developer will want to use it inside INAV in some way that they think makes sense for a "pilot name", perhaps to create a unique name for a configuration file `%pilot_name%-%craft_name%-%date%.cfg` or the like. I want to avoid this, as it is precisely the reason Craft Name cannot be used to solve my use case.

**Comment by aubreyrjones** [comment]
Alright, rebase and all suggested edits complete.

**Comment by aubreyrjones** [comment]
So... if I rebase, any chance I could get this merged soon?

**Comment by shellixyz** [comment]
Again I'm not taking the decisions to merge or not but there is now the support for string values you should change your code to use those. Will make your code simpler and the code size smaller.

**Comment by aubreyrjones** [comment]
That's crazy. This patch has been in a merge-ready state twice and was simply ignored. I have no reason to believe that if I update it, it would be merged. So right now, it mostly seems that if I updated it, then the Real INAV Developers™ will implement quantum tunneled settings instead of pressing the "merge" button and I'll need to rewrite again.

INAV developers say on Facebook and Telegram that they need more developers... but this is really, really discouraging. I want to contribute to the mixer to support asymmetrical control surfaces. But if this trivial patch is being ignored, why should I believe any more substantive contribution would be merged?

**Comment by digitalentity** [comment]
@aubreyrjones at the moment you seem to be the only person who wants this setting. While I deem it useful in some cases, I won't defer other changes waiting for for this one. I will, however, merge this PR once it doesn't have any merge conflicts and updates relevant documentation.

**Comment by aubreyrjones** [comment]
@digitalentity Great! I'm not asking for you to delay or change anything at all. I'm just asking if I meet you on your schedule, at a synchronization point of your choosing, on your conditions, that I can have some reasonable expectation that it will be merged soon enough that I don't have to re-do it again.

Would you like this re-written with a string setting? I agree that it would be cleaner, and I went looking for a string setting when I first implemented this. The rebase looks extensive enough that I don't care which way I do it this time.

For documentation, I assume you mean the CLI docs? Is there another place this is documented? (I already include tooltips in the accompanying configurator patch.)

**Comment by digitalentity** [comment]
Yes, re-implementation using sting setting would be cleaner. By docs I mean Cli.md.

**Comment by aubreyrjones** [comment]
@digitalentity Awesome! Thanks. I'll get on this soon.

**Comment by DzikuVx** [comment]
@aubreyrjones could you please rebase this on latest `development` branch?

**Comment by fiam** [comment]
Moving to 2.2. Might be superseded by pilot name if no objections are raised.

**Comment by aubreyrjones** [comment]
I'm closing this pull request. I've lost my interest in contributing to inav.

**Pull request closed without merging**
